### DISCUSSION TITLE
To cross-compile or not to cross-compile

### DISCUSSION THREAD

**Discussion by tlb-tarek-sater** [bevyengine/bevy#11752] [Q&A]
I'm aiming to target the 3 desktop platforms (Windows, Mac, Linux) for a bevy project. I know that Rust can cross-compile from Linux to all these platforms, but should I?

Is there any performance impact to doing so? Or maybe some other gotchas compared to compiling on the native machine?

Would love to hear your thoughts.

**Comment by BorisBoutillier** [comment]
The easiest is to use a pre-made template that contains CI scripts to cross-compile for you on different platform.
A good example is :
https://github.com/bevyengine/bevy_github_ci_template

It even supports WASM builds and push all releases target to an itch.io project.

**Reply by tlb-tarek-sater** [reply]
I'm aware of the CI script, I'm just asking if there is a performance impact or other gotchas to doing cross-compilation rather than building on the native OS.

**Comment by rlidwka** [comment]
If you can cross-compile, you should cross-compile. Making all artifacts in one place is less prone to errors, so there's no reason not to do it.

Only question is whether you can cross-compile. Bevy had some issues with blake3 crate previously, but those are probably solved. If you have other FFI deps, you might have issues (as I do with libcurl).

**Comment by afonsolage** [comment]
AFAIK there is no runtime performance difference between native and cross-compiled binaries.
